components/properties/src/script.rs

```diff
@@ -140,6 +148,58 @@ impl<'data> ScriptExtensions<'data> {
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Returns whether `script` satisfies either of: 1) matches the Script
+    /// property value; 2) is contained in the Script_Extensions property value
+    pub fn has_script(&self, code_point: u32, script: Script) -> bool {
+        if script == self.get_script_val(code_point) {
+            true
+        } else {
+            let scx_val = self.get_script_extensions_val(code_point);
+            let script_find = scx_val.iter().find(|&sc| sc == script);
+            script_find.is_some()
+        }
+    }
+
+    /// Returns a [`UnicodeSet`] for the given [`Script`] which represents all
+    /// code points for which `has_script` will return true.
+    pub fn get_script_extensions_set(&self, script: Script) -> UnicodeSet {
+        let mut builder = UnicodeSetBuilder::new();
+        let mut range_start: u32 = 0;
+
+        loop {
+            let next_cpm_range = self.trie.get_range(range_start);
+
+            match next_cpm_range {
+                Some(cpm_range) => {
+                    let end = cpm_range.get_end();
+                    if end <= CODE_POINT_MAX {
+                        let start = cpm_range.get_start();
+
+                        if self.has_script(start, script) {
+                            builder.add_range_u32(&(start..=end));
+                        }
```

Comment:
This works, but has_script() does a trie lookup which is unnecessary because get_range() already gave you the trie value. Faster if you call a lower-level impl function with the trie value (or copy/inline-duplicate a couple of lines).